app/services/account_reset/validate_granted_token.rb

```diff
@@ -0,0 +1,27 @@
+module AccountReset
+  class ValidateGrantedToken
+    include ActiveModel::Model
+    include GrantedTokenValidator
+
+    def initialize(token)
+      @token = token
+    end
+
+    def call
+      @success = valid?
+
+      FormResponse.new(success: success, errors: errors.messages, extra: extra_analytics_attributes)
```

Comment:
Not fond of a service returning a form response.

Reply:
This is not the first time we have a service returning a FormResponse. Perhaps the issue is with the naming of the response class? If we think there's a better name that can apply to both service and form objects, then can certainly discuss that later. Maybe drop the "form" and call it `Response`?

Reply:
A new name would be an improvement but it would also be a lot of work and not worth the effort at this point.